js/controllers/todos_controller.js

```diff
@@ -1,4 +1,125 @@
-App.TodosController = Ember.ObjectController.extend({
+
+Abs.TasksController = Ember.ObjectController.extend({
     actions: {
+        editTasks: function () {
+            this.set('editMode', true);
+        },
+        changeTask: function () {
+            var me = this;
+            Ember.run.later((function () {
+                me.set('editMode', false);
+                if (Ember.isEmpty(me.get('model.title'))) {
+                    me.send('removeTask');
+                } else {
+                    me.get('model').save();
+                }
+            }), 100);
+        },
+        removeTask: function () {
+            var Task = this.get('model');
+            Task.deleteRecord();
+            Task.save();
+        }
+    },
+
+    editMode: false,
+
+    isDone: function (key, value) {
+        var model = this.get('model');
+
+        if (value === undefined) {
+            // property being used as a getter
+            return model.get('isDone');
+        } else {
+            // property being used as  setter
+            model.set('isDone', value);
+            model.save();
+            return value;
+        }
+    }.property('model.isDone')
+});
+
+Abs.AbsController = Ember.ArrayController.extend({
+    actions: {
+        checkAll: function () {
+            var checked = $('#chk-all')[0].checked;
+            var store = this.get('store'), model = this.get('model');
+
+            model.forEach(function (task) {
+                task.set('isDone', checked);
+                task.save();
+            });
+
+        },
+        createTask: function () {
+            // Get the Tasks title set by the "New Tasks" text field
+            var title = this.get('newTitle');
+            if (!title.trim()) { return; }
+
+            // Create the new Tasks model
+            var Tasks = this.store.createRecord('Tasks', {
+                title: title,
+                isDone: false
+            });
+
+            // Clear the "New Tasks" text field
+            this.set('newTitle', '');
+
+            // Save the new model
+            Tasks.save();
+        },
+
+        removeAllCompleted: function () {
+            var model = this.get('model');
+            model.forEach(function (task) {
+                if (task && task.get('isDone')) {
+                    Ember.run.once(this, function () {
+                        task.deleteRecord();
+                        task.save();
+                    });
+                }
+
+            });
+
+        },
+
+        allTasks: function () {
+            var me = this, store = me.get('store');
```

Comment:
I usually create a variable of "this" because this changed in different scope, and me is shorter than this. however in this method it's not necessary 


     On Monday, September 28, 2015 5:58 PM, Ruan Naude <notifications@github.com> wrote:
   

 In js/controllers/todos_controller.js:> +        removeAllCompleted: function () {
> +            var model = this.get('model');
> +            model.forEach(function (task) {
> +                if (task && task.get('isDone')) {
> +                    Ember.run.once(this, function () {
> +                        task.deleteRecord();
> +                        task.save();
> +                    });
> +                }
> +
> +            });
> +
> +        },
> +
> +        allTasks: function () {
> +            var me = this, store = me.get('store');
why create the var me?—
Reply to this email directly or view it on GitHub.